engine/src/main/java/org/terasology/persistence/typeHandling/SimpleTypeHandler.java

```diff
@@ -30,9 +31,7 @@
     @Override
     public PersistedData serializeCollection(Collection<T> value, SerializationContext context) {
         List<PersistedData> rawList = Lists.newArrayList();
-        for (T item : value) {
-            rawList.add(serialize(item, context));
-        }
+        rawList.addAll(value.stream().map(item -> serialize(item, context)).collect(Collectors.toList()));
```

Comment:
I'd go with the original version, because I find it easier to read and it's called very often.